src/pages/settings/email-settings/EmailSettings.tsx

```diff
@@ -93,6 +163,54 @@ export function EmailSettings() {
           />
         </Element>
 
+        <Element leftSide={t('enable_e_invoice')}>
+          <Toggle
+            checked={company?.settings.enable_e_invoice}
+            onValueChange={(value) =>
+              handleChange('settings.enable_e_invoice', value)
+            }
+          />
+        </Element>
+        {company?.settings.enable_e_invoice === true ? (
+          <>
+            <Element
+              leftSide={t('upload_certificate')}
+              leftSideHelp={company?.has_e_invoice_certificate ? t('certificate_set') : t('certificate_not_set')}
+            >
+              <div
+                {...getRootProps()}
+                className="flex flex-col md:flex-row md:items-center"
+              >
+                <div className="relative block w-full border-2 border-gray-300 border-dashed rounded-lg p-12 text-center hover:border-gray-400 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500">
+                  <input {...getInputProps()} />
+                  <Image className="mx-auto h-12 w-12 text-gray-400" />
+                  <span className="mt-2 block text-sm font-medium text-gray-900">
+                    {isDragActive
+                      ? 'drop_your_logo_here'
+                      : t('dropzone_default_message')}
+                  </span>
+                </div>
+              </div>
+            </Element>
+
+            <Element
+              leftSide={t('certificate_passphrase')}
+              leftSideHelp={company?.has_e_invoice_certificate_passphrase ? t('passphrase_set') : t('passphrase_not_set')}
+            >
+              <InputField
+                value={''}
+                id="password"
+                type="password"
+                onValueChange={(value) =>
+                  handleChange('has_e_invoice_certificate_passphrase', value)
+                }
+              />
+            </Element>
+          </>
+        ) : (
+          ''
```

Comment:
If you decide not to go with && approach, previously mentioned, at least put `null` here instead of empty string, creates one less node.